presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestNestedFieldConverter.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.facebook.presto.common.type.ArrayType;
+import com.facebook.presto.common.type.RowType;
+import com.facebook.presto.common.type.StandardTypes;
+import com.facebook.presto.common.type.TypeManager;
+import com.facebook.presto.common.type.TypeSignatureParameter;
+import com.google.common.collect.ImmutableList;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.common.type.DateType.DATE;
+import static com.facebook.presto.common.type.DecimalType.createDecimalType;
+import static com.facebook.presto.common.type.DoubleType.DOUBLE;
+import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static com.facebook.presto.common.type.RealType.REAL;
+import static com.facebook.presto.common.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.common.type.VarcharType.createUnboundedVarcharType;
+import static com.facebook.presto.iceberg.NestedFieldConverter.toIcebergNestedField;
+import static com.facebook.presto.iceberg.NestedFieldConverter.toPrestoNestedField;
+import static com.facebook.presto.metadata.FunctionAndTypeManager.createTestFunctionAndTypeManager;
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class TestNestedFieldConverter
+{
+    @DataProvider(name = "allTypes")
+    public static Object[][] testAllTypes()
+    {
+        return new Object[][] {
+                {1, "boolean"},
+                {1, "integer"},
+                {1, "array"},
+                {1, "bigint"},
+                {1, "real"},
+                {1, "double"},
+                {1, "map"},
+                {1, "decimal"},
+                {1, "varchar"},
+                {1, "varbinary"},
+                {1, "row"},
+                {1, "date"}
+        };
+    }
+
+    @Test(dataProvider = "allTypes")
+    public void testToPrestoNestedField(int id, String name)
```

Comment:
Also add a negative test case for the anonymous row field type

Reply:
Maybe also add some tests for real nested types, for example:

```
row(a int, b varchar, c array(row(d int)), e map(varchar, row(f varchar)))
```

Reply:
@aditi-pandit @hantangwangd I have updated the PR with more tests.